python/pyarrow/tests/test_table.py

```diff
@@ -1746,3 +1746,29 @@ def test_table_select():
     result = table.select(['f2'])
     expected = pa.table([a2], ['f2'])
     assert result.equals(expected)
+
+
+def test_table_group_by():
+    table = pa.table([
+        pa.array([1, 2, 3, 4, 5]),
+        pa.array(["a", "a", "b", "b", "c"]),
+    ], names=["values", "keys"])
+
+    r = table.group_by("keys", ["values"], "hash_sum")
```

Comment:
Should we hide the "hash_" prefix as an implementation detail? @lidavidm @jorisvandenbossche

Reply:
Yes, I would also hide the "hash_" here (this is also not a compute-module function you are calling, so I think it's fine that there is some deviation)